### PR TITLE
Code Insights: Add template section to the getting started page

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/30158

## Context
[Designs](https://www.figma.com/file/EFgTP7gpdpQSEo1n4FUfsW/In-product-landing-page-WIP?node-id=1128%3A9814)

This PR just adds the template section component to the getting started page. Also in these PR changes, you can find a static non UI file where all templates for code insights and this section are stored. If you want to add/delete/edit a template for this section this is your file to edit. 

Screenshot 2022-01-31 at 18 43 50

## For reviewers 

@oleggromov I had to slightly extend the `<Tab />` component in the wildcard in order to add custom styles.

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/getting-started/components/code-insights-temlates/constants.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+interface CaptureGroupTemplate {
+    type: InsightType.CaptureGroup
+    title: string
+    description: string
+    templateValues: Partial<CaptureInsightUrlValues>
+}
+
+export const TEMPLATE_SECTIONS: TemplateSection[] = [
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/pages/getting-started/components/code-insights-temlates/constants.ts:26]: @AlicjaSuska @Joelkw @felixfbecker @coury-clark @CristinaBirkel @unclejustin if you have ideas around these templates and you want to add (or remove some templates from the template section) please feel free to edit this list of templates.

**Review by unclejustin — APPROVED** [review]

**Comment by unclejustin** [comment]
@vovakulikov LGTM 🌮  but you've got an eslint error in your build

**Comment by vovakulikov** [comment]
@unclejustin thanks for pointing this out. I almost forgot to ask @AlicjaSuska @Joelkw have we already got a doc page for code insights use cases? Are the following two links are different? 

Screenshot 2022-01-31 at 19 56 10

Screenshot 2022-01-31 at 19 56 15

**Comment by Joelkw** [comment]
@AlicjaSuska  was your intention these go to the existing common use cases page here: https://docs.sourcegraph.com/code_insights/references/common_use_cases ?

**Pull request merged**